examples/extending/extending-leaflet.md

```diff
@@ -0,0 +1,445 @@
+---
+layout: tutorial_v2
+title: Extending Leaflet
+---
+
+## Extending Leaflet
+
+Leaflet has literally hundreds of plugins. These expand the capabilities of Leaflet: sometimes in a generic way, sometimes in a very use-case-specific way.
+
+Part of the reason there are so many plugins is that Leaflet is easy to extend. This tutorial will cover the most commonly used ways of doing so.
+
+Please note that this tutorial assumes you have a good grasp of:
+
+* [JavaScript](https://developer.mozilla.org/en-US/Learn/JavaScript)
+* [DOM handling](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction)
+* [Object-oriented programming](https://en.wikipedia.org/wiki/Object-oriented_programming) (understanding concepts like classes, instances, inheritance, methods and properties)
+
+
+## Leaflet architecture
+
+Let's have a look at a simplified UML Class diagram for Leaflet 1.0.0. There are more than 60 javascript classes, so the diagram is a bit big. Luckily we can make a zoomable image with a `L.ImageOverlay`:
+
+{% include frame.html url="class-diagram.html" %}
+
+
+From a technical point of view, Leaflet can be extended in different ways:
```

Comment:
I would only list the 3 top level li-s here, not the 6 second level ones, as this gets explained later in more detail.